src/Kafka/KafkaMessageBrowser/components/FilterGroup.tsx

```diff
@@ -0,0 +1,145 @@
+    timestamp: t("filter.timestamp"),
+    epoch: t("filter.epoch"),
+    latest: t("filter.latest"),
+  };
+  return (
+    <ToolbarItem>
+      <InputGroup>
+        <Dropdown
+          toggle={
+            <DropdownToggle
+              onToggle={setIsOpen}
+              isDisabled={isDisabled}
+              data-testid={"filter-group"}
+            >
+              {labels[currentCategory]}
+            </DropdownToggle>
+          }
+          position={DropdownPosition.left}
+          isOpen={isOpen}
+          onSelect={() => setIsOpen(false)}
+          dropdownItems={[
+            <DropdownItem
+              key="offset"
+              value="offset"
+              autoFocus={currentCategory === "offset"}
+              onClick={() => setCurrentCategory("offset")}
+            >
+              {labels["offset"]}
+            </DropdownItem>,
+            <DropdownItem
+              key="timestamp"
+              value="timestamp"
+              autoFocus={currentCategory === "timestamp"}
+              onClick={() => setCurrentCategory("timestamp")}
+            >
+              {labels["timestamp"]}
+            </DropdownItem>,
+            <DropdownItem
+              key="epoch"
+              value="epoch"
+              autoFocus={currentCategory === "epoch"}
+              onClick={() => setCurrentCategory("epoch")}
+            >
+              {labels["epoch"]}
+            </DropdownItem>,
+            <DropdownSeparator key={"divider"} />,
+            <DropdownItem
+              key="latest"
+              value="latest"
+              autoFocus={currentCategory === "latest"}
+              onClick={() => {
+                setCurrentCategory("latest");
+                onLatest();
+              }}
+            >
+              {labels["latest"]}
+            </DropdownItem>,
+          ]}
+        />
+        {currentCategory === "offset" && (
+          <TextInput
+            isDisabled={isDisabled}
+            type={"number"}
+            aria-label={t("filter.offset_aria_label")}
+            placeholder={t("filter.offset_placeholder")}
+            onChange={(value) => {
+              if (value !== "") {
+                const newOffset = parseInt(value, 10);
+                if (Number.isInteger(newOffset)) {
+                  onOffsetChange(newOffset);
+                }
+              } else {
+                onOffsetChange(undefined);
+              }
+            }}
+            value={offset || ""}
+          />
+        )}
+        {currentCategory === "timestamp" && (
+          <DateTimePicker
+            isDisabled={isDisabled}
+            value={timestamp ? timestamp * 1000 : undefined}
+            onChange={onTimestampChange}
+          />
+        )}
+        {currentCategory === "epoch" && (
+          <TextInput
+            isDisabled={isDisabled}
+            type={"number"}
+            aria-label={t("filter.epoch_aria_label")}
+            placeholder={t("filter.epoch_placeholder")}
```

Comment:
Playing around with classes currently provided, one way to modify the width of an input within the `<InputGroup>` component is to apply the following attributes and values:
`<TextInput className="pf-u-flex-basis-auto pf-u-flex-grow-0 pf-u-w-initial" size="{length of placeholder text}...`

Since InputGroup is display:flex, we can use the flex utility classes to modify how this element displays so that with basis:auto and grow:0, the input can be the size it wants to be.

Then the `size` attribute can dictate how long the input should be.

If we can specify a size to be based on the length of the placeholder text, that should result in a text input that is long enough to display the placeholder text.